pkg/reconciler/internal/updater/updater_test.go

```diff
@@ -68,12 +85,12 @@ var _ = Describe("Updater", func() {
 
 	When("an update is a change", func() {
 		It("should apply an update function", func() {
-			u.Update(EnsureFinalizer(testFinalizer))
+			u.Update(EnsureAnnotation("foo", "bar"))
```

Comment:
Maybe we can use `RemoveFinalizers` for this test (which I think would require manually putting a finalizer on the object)?

Or, if that would complicate the test setup too much, I suppose we could keep `EnsureAnnotation` around, but I would unexport it and change it to be _much_ simpler. Something like:
```go
func setAnnotationsFooBar() UpdateFunc {
	return func(obj *unstructured.Unstructured) bool {
		obj.SetAnnotations(map[string]string{"foo":"bar"})
		return true
	}
}
```